magento/magento2
# Fix price indexer for Configurable product with out of stock children

### Description (*)
If a child product is out of stock, but out of stock products shall be shown, this check removes the child from the price indexer.
This way the min_price of the configurable is calculated as 0 if it has only childs, which are out of stock, even if they shall be shown.
So the price filter will not work correctly in this case.
That is why the check should be the other way round. The filter should only be applied, if out of stock products shall NOT be shown and so also not be used for the price calculation of the configurable.
See https://github.com/magento/magento2/commit/72f2b8a22d1cae2f023559e51a2fb88080c48a98#r33460279

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#27341 : Category Price filter does not work for configurables with out of stock childs, when ShowOutOfStock Option is enabled

### Manual testing scenarios (*)
1. Create an configurable product with two child products with a price of 100$ resp. 200$ and add the configurable to a category
2. Set the cheaper child product in the advanced inventory to manage stock -> true, but to be not in stock and save the product
3. Set cataloginventory/options/show_out_of_stock to YES
4. Set the cheaper child product in the advanced inventory to manage_stock -> false (this does not change the value of is_in_stock in the database. This remains 0)
5. Go to the category and see, that the price for your configurable is displayed as starting from 100 $, as out of stock products shall be displayed according to your settings.
6. Try to filter for a range which includes 100$
7. You would assume to see the configurable in the filtered collection, but you don't.

### Questions or comments


### Contribution checklist (*)
